### DISCUSSION TITLE
Noob Frigate+ Model Question

### DISCUSSION THREAD

**Discussion by domenicdistefano** [blakeblackshear/frigate#7977] [Ask A Question]
This question is like when you ask ChatGPT to respond like you are a five year old... :)

I just received my invitation & completed the registration for Frigate+, upgraded to the .13 beta, and added my API key to my Docker Compose file as per [this page](https://deploy-preview-6262--frigate-docs.netlify.app/plus/) noted in the welcome email. 

So if I am understanding how all of this works correctly Frigate+ offers what I am going to call an "upgraded baseline model" that is superior to what we have been using pre-Frigate+ in that its based & trained on Frigate community submissions  and therefore should be more accurate with detections for a subset of the currently included object types which will expand over time. Also license plate for example is new but will not trigger a detection as its a subset of car. If I got all of that right do I need to actually do anything other than entering my API key into the docker compose file and firing everything up? If so how do I verify my API key was properly recognized and its using the Frigate+ baseline model as I am calling it?

So now after the above is taken care of my further understanding is that I can submit my own Frigate+ detections (lets say my regular UPS or FedEx truck) to both improve the model everyone else with Frigate+ is using overall in addition to making my local install a little bit smarter for my particular set of cameras. This is where my understanding decreases somewhat. So now that I have submitted one or more of my own detections it seems this is where I then need to use one of the credits. 1 credit = as many detections as I submit over time (which is one "custom" model specific to me)? or perhaps 1 credit = one detection I submit? In either case now that I have submitted custom detections / used a credit how exactly does that work in terms of how I need to configure Frigate so that everything works?

Seems like you need a page in your docs similar to the "[What does Frigate use the CPU for and what does it use a detector for? (ELI5 Version)](https://docs.frigate.video/frigate/hardware/#what-does-frigate-use-the-cpu-for-and-what-does-it-use-a-detector-for-eli5-version)"

Thank you for all of the hard  work with this awesome project!

**Comment by blakeblackshear** [comment]
The base model isn't available directly. You will need to start by uploading, annotating, and verifying at least 10 images to start. Then you will request your first model which will use one credit. Once your model is finished training, you will [update your configuration](https://deploy-preview-6262--frigate-docs.netlify.app/integrations/plus#use-models) to use the model id from your Frigate+ account.

As needed, you can submit additional images to Frigate+. Once you have added at least 10 more verified images (more is [recommended](https://deploy-preview-6262--frigate-docs.netlify.app/plus/#improving-your-model)), then you can submit a request to train an updated model with the additional examples. Each model request will generate a new model ID and will use one credit regardless of how many images you upload and verify.

**Comment by domenicdistefano** [comment]
Thank you - the five year old in me understands much better now!

So beyond a couple / few uploads of UPS & FedEx trucks any guidance for what types of additional submissions makes the most sense for the resulting model to be best tailored to my cameras? Is the idea / purpose to solve for situations where for instance a grill I have on my back patio currently is constantly being detected as a a person despite no matter what I do with motion & zone masks? Same things with a hose reel I have. How about my two dogs that get detected most of the time as people? Should I submit pictures of my specific dogs tagged as dogs so going forward it knows my dogs are dogs? Is the criteria of labels specific to the few object types you list as being currently supported?

Specific question about car / license plates as a label type (could also apply to people / faces I guess). In my case three of my 12 cameras are specialized / $1K each LPR cameras that actually have onboard ASICs that already do a great job in identifying, OCRing, cataloging plates and can send emails when specific plates are identified - not trying to reinvent this functionality with Frigate. My question is what if I still want to see what Frigate+ can do with this flood of car & plate data I could still throw at it? Would it be beneficial to the community for me to submit generic plates so the general model is improved? What  about the specific submissions of just my car / plate in my driveway which like the grill and hose reel routinely get recognized as a person or dog? Also when detecting my specific vehicles I could then send alerts / triggers to HA as the cameras themselves do not interface with anything else so I can't use them currently with HA.

Sorry for all of the questions but all of this is blowing my mind on the potential of what I can do going forward.  

Bonus Question - Up until now as I have been experimenting with my camera detect resolutions I frequently delete my storage folder and allow Frigate to completely rebuild its database so it comes up as a fresh install in between major config changes on my part. Going forward now as make submissions to Frigate+ from the UI will my practice of doing this cause "break" my submissions before and / or after I request and receive my custom model(s)? Main reason I do this is because I haven't found a "bulk delete all detections" anywhere so I have just been nukeing everything.

**Reply by NickM-27** [reply]
> So beyond a couple / few uploads of UPS & FedEx trucks any guidance for what types of additional submissions makes the most sense for the resulting model to be best tailored to my cameras?

You will want to provide all kinds of examples, see this [excerpt from the docs](https://deploy-preview-6262--frigate-docs.netlify.app/plus/#improving-your-model)

> In order to request your first model, you will need to have annotated and verified at least 10 images. Each subsequent model request will require that 10 additional images are verified. However, this is the bare minimum. For the best results, you should provide at least 100 verified images per camera. Keep in mind that varying conditions should be included. You will want images from cloudy days, sunny days, dawn, dusk, and night.
>
> As circumstances change, you may need to submit new examples to address new types of false positives. For example, the change from summer days to snowy winter days or other changes such as a new grill or patio furniture may require additional examples and training.

The more variety that you can provide, the more circumstances the model will be tuned to your cameras for.

> Is the idea / purpose to solve for situations where for instance a grill I have on my back patio currently is constantly being detected as a a person despite no matter what I do with motion & zone masks? Same things with a hose reel I have. How about my two dogs that get detected most of the time as people? Should I submit pictures of my specific dogs tagged as dogs so going forward it knows my dogs are dogs? Is the criteria of labels specific to the few object types you list as being currently supported?

I'd suggest reading the [docs on labeling images](https://deploy-preview-6262--frigate-docs.netlify.app/plus/#properly-labeling-images)

yes, in frigate 0.13 there is the option to mark these as `this is not a dog` or `this is not a person` and then you should go into frigate+ and annotate the image including labeling the object as what it actually is, then verify the image. This will teach the model that these things are not a person, dog, etc. and after enough examples it will stop detecting it altogether. This has worked really well for me and the false positives that I used to have. 

> Specific question about car / license plates as a label type (could also apply to people / faces I guess). In my case three of my 12 cameras are specialized / $1K each LPR cameras that actually have onboard ASICs that already do a great job in identifying, OCRing, cataloging plates and can send emails when specific plates are identified - not trying to reinvent this functionality with Frigate. My question is what if I still want to see what Frigate+ can do with this flood of car & plate data I could still throw at it? Would it be beneficial to the community for me to submit generic plates so the general model is improved?

I don't know the answer to this, will need to see what Blake says

> What about the specific submissions of just my car / plate in my driveway which like the grill and hose reel routinely get recognized as a person or dog?

This is no different than the above example as far as I can tell

> Bonus Question - Up until now as I have been experimenting with my camera detect resolutions I frequently delete my storage folder and allow Frigate to completely rebuild its database so it comes up as a fresh install in between major config changes on my part. Going forward now as make submissions to Frigate+ from the UI will my practice of doing this cause "break" my submissions before and / or after I request and receive my custom model(s)? Main reason I do this is because I haven't found a "bulk delete all detections" anywhere so I have just been nukeing everything.

Once you submit the example to frigate+ the image on your host can be deleted and it does not make a difference, naturally if you delete the database or the image file before uploading to Frigate+ then you won't be able to do so from the frigate webUI

**Reply by blakeblackshear** [reply]
> Would it be beneficial to the community for me to submit generic plates so the general model is improved?

It depends on whether or not the camera's view of a car and it's license plate look similar to the way they do for other users. It is really only license plate detection, not recognition. For example, if you have an LPR camera that is mounted low to the ground and positioned to just get a great shot of a plate, then probably not that useful for improving the model generally.

**Comment by domenicdistefano** [comment]
Day / night samples of one of my three LPR cameras below - all are on a 10' high 4x4s and take similar views of cars & plates as they go by ~250K per year. 

Currently I have car detection turned off in Frigate due to the volume of vehicles & I get the plates & OCR through the camera OS (more interested personally in people walking their dogs or amazon / ups / fedex deliveries detected with Frigate).

If samples of these would be beneficial to the model I can upload as many as you like to Frigate+ through all four seasons day & night. I think the docs say you can't trigger on a plate? My ultimate goal is to not capture 200K vehicles per year with Frigate obviously. The functionality with Frigate I would like to see is for it to send an alert to HA when only a ups  fedex / amazon truck is detected. Once in HA I can create a rule to then do something special if I am expecting a package on that day for instance.

![image](https://github.com/blakeblackshear/frigate/assets/7042586/22a34228-1252-4bb9-900b-ab043115d191)

![image](https://github.com/blakeblackshear/frigate/assets/7042586/a0cff8f6-05fb-4f0e-a57a-774a6e637de5)
![image](https://github.com/blakeblackshear/frigate/assets/7042586/78533d1e-187f-4de4-8757-b77522b7f3e7)

**Reply by NickM-27** [reply]
>  The functionality with Frigate I would like to see is for it to send an alert to HA when only a ups fedex / amazon truck is detected. Once in HA I can create a rule to then do something special if I am expecting a package on that day for instance.

With Frigate+ you can already do this today, to be clear

**Reply by domenicdistefano** [reply]
Excellent! Must have misunderstood / misread a section of the docs. Lots to digest.

**Reply by blakeblackshear** [reply]
I think these examples would be helpful. No need to submit thousands, but submitting a few periodically would be great.
